internal/aws/cwlogs/pusher_test.go

```diff
@@ -114,6 +114,7 @@ func newMockPusher() *logPusher {
 	return newLogPusher(StreamKey{
 		LogGroupName:  logGroup,
 		LogStreamName: logStreamName,
+		Entity:        &entity,
```

Comment:
does this have to be a pointer? That is what is breaking your comparability

Reply:
If you have a comparable version of entity w/o string pointers then you wouldn't need the hash.  You could then convert it to the output entity type in getPusher when you create the new pusher

Your "key" type could just have the fields that need to be unique or something like that.  It doesn't need to be a dict I struct of a map, you could do whatever you want with the format, maybe something similar to what you are ultimately being forced to do in your hash method

Reply:
Currently kept the PR as is, as using a Hash of the StreamKey as the key for the pusher map is not affecting the functionality. But I will try to see if there's a way we can "pack" the entity to a comparable type, and "unpack" when making the PLE call.